src/Application/Actions/Hooks/StripeChargeUpdate.php

```diff
@@ -214,6 +234,14 @@ private function handleChargeRefunded(Event $event, Response $response): Respons
 
         $isTipRefund = $donation->getTipAmountFractional() === $amountRefunded;
         $isFullRefund = $donation->getAmountFractionalIncTip() === $amountRefunded;
+        // If things disagree about the original captured amount, this is still uncharted
+        // territory and we should refrain from processing. If that matches but there is an
+        // over-refund, we'll want to notify the team but we know we are best off handling it
+        // like a simple refund from a donation status perspective.
+        $isOverRefund = (
+            $donation->getAmountFractionalIncTip() === $charge->amount_captured &&
+            $donation->getAmountFractionalIncTip() < $amountRefunded
```

Comment:
IMHO easier to read this way - we think of it as an over-refund, not an under-donation:
```suggestion
             $amountRefunded > $donation->getAmountFractionalIncTip()
```